### DISCUSSION TITLE
LFX Mentorship '23 Term 02 June - August

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#937] [Announcements]
Hello from KubeCon EU everyone!

The LFX Mentorship 2023 Term 02 June - August is open for project ideas:
https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug

Project submission and application timeline:
* project proposals due: **May 9**, 2023
* mentee applications open: **May 10 - 23**, 2023
* application review/admission decisions/HR paperwork: **May 24 - 29**, 2023

If your Google Summer of Code proposal was declined, this is a good alternative -- proposals can be copied over almost 1:1.

We're looking forward to seeing all the project ideas you're interested in working on!

### Timeline

Mentorship duration - three months (12 weeks - full-time schedule)

| activity | date |
| --- | --- |   
| project proposals due | Tue, May 9, 5:00 PM PDT |
| mentee applications open | Wed May 10 - Tue May 23, 5:00 PM PDT |
| application review/admission decisions | Wed May 24 - Mon May 29, 5:00 PM PDT |
| Mentorship program begins with the initial work assignments |  Thur June 1 (Week 1) | 
| Midterm mentee evaluations and first stipend payments | Wed July 12 (Week 6) |
| Final mentee evaluations and mentee feedback/blog submission due, second and final stipend payment approvals | Wed Aug 23, 5:00 PM PST (Week 12) |
| Last day of term | Thur Aug 31 |

---

[Announcement email](https://lists.cncf.io/g/tag-cs-mentoring-wg/message/8)

**Comment by MahadMuhammad** [comment]
Hello, @nate-double-u! I am interested in applying for the `LFX Mentorship '23 Term 02 June - August` program during the summer (assuming I am not accepted into GSoC). I was wondering if there are any project ideas that have been selected on your end. I am eager to learn and contribute to LFX :octocat:

**Reply by nate-double-u** [reply]
Hi @MahadMuhammad, Thanks for your interest! Currently we're still looking for project proposals. Mentee applications open May 5th. If you're looking for more information on how to participate, I'd suggest looking at this getting started https://contribute.cncf.io/contributors/getting-started/ article.

**Reply by MahadMuhammad** [reply]
Thanks @nate-double-u for letting me know :)

**Comment by nate-double-u** [comment]
Please note that because of KubeCon, and Google Summer of Code's announcement window, we've extended the project proposals period! (and shifted other dates accordingly)

Project proposals are now due **May 9, 2023**.

**Comment by razvand** [comment]
Hi, @nate-double-u. What projects can apply for the LFX Mentorship? They must be CNCF-approved projects?

**Reply by nate-double-u** [reply]
Hi @razvand, this program is for CNCF projects at any level (sandbox, incubating or graduated).

**Comment by satyampsoni** [comment]
HI @nate-double-u. I came to know about this program today only. Can I still apply for it and what are the requirements to fulfill for this program.

**Reply by nate-double-u** [reply]
Hi @satyampsoni, the program is in it's proposals phase right now. The project maintainers are currently submitting ideas for mentes to work on. Next week mentee applications will be opened -- you'll see instructions and links to apply on the [Term 02 - 2023 June - August](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug) page.

For information on getting started I recommend reading [our page for mentees](https://github.com/cncf/mentoring/tree/main/mentees), and _[Do you want to start contributing to open source and need help figuring out where to begin?](https://contribute.cncf.io/contributors/getting-started/)_

**Comment by rukundob451** [comment]
Hello @nate-double-u, been keeping an eye for this program through and currently waiting for the mentee applications to be opened. Hoping I can be a part of the [Term 02 - 2023 June - August](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug) that is incoming.

**Comment by satyampsoni** [comment]
As a mentee will I be asked to write the proposal?

**Reply by haanhvu** [reply]
You mean as an applicant?

To apply for the LFX mentorship program, you just need to submit your resume and cover letter. However, some orgs may later check your understanding/suitability for the project through interviewing, asking you further questions, or asking you to send a project proposal.

**Comment by nate-double-u** [comment]
A friendly reminder for project maintainers, CNCF LFX mentorship term 2 project proposals are due tomorrow! (Tue, May 9, at 5:00 PM PDT)

**Comment by antonio-pedro99** [comment]
Hello @nate-double-u, I just want to make sure the mentee applications are now open.

**Reply by aliok** [reply]
Hi @antonio-pedro99 ! They are not open yet.

@nate-double-u is working on it. We will write a message here on this discussion entry when they're open.

**Reply by aliok** [reply]
Applications are now open: https://github.com/cncf/mentoring/discussions/988 🎉

**Comment by rukundob451** [comment]
Hello @nate-double-u kindly confirm whether mentee applications are now open?

**Reply by aliok** [reply]
Hi @rukundob451 ! They are not open yet.

@nate-double-u is working on it. We will write a message here on this discussion entry when they're open.

**Reply by nate-double-u** [reply]
Thanks @aliok, this is true, I'm going through the process, and the next step is to get LFX approval. I'm hoping this will be mostly done today. I will make an announcement when it's all ready, thanks for the patience, and I'm sorry for the delay.

**Reply by satyampsoni** [reply]
Thanks for informing and putting so much efforts in the backend.

**Reply by aliok** [reply]
Hello,

Applications are now open: https://github.com/cncf/mentoring/discussions/988 🎉

**Comment by aliok** [comment]
Thanks for the project ideas you've proposed as a mentor!

Applications for mentees are now open: https://github.com/cncf/mentoring/discussions/988

Please inform your community about the LFX Mentorship opportunities and expect mentee candidates looking to engage with you and your community.

**Reply by aliok** [reply]
@Abhishek-kumar09 @Ankurk99 @FeynmanZhou @GiedriusS @MariamFahmy98 @ReToCode @TerryHowe @Vishal-Chdhry @aliok @amityt @arkajyotiMukherjee @asmitbm @aufi @barnettZQG @caniszczyk @chipzoller @chivalryq @chrisohaver @craigbox @daemon1024 @djzager @dwertent @eddycharly @fouioui

**Reply by aliok** [reply]
@fouioui @gyohuangxin @hydai @ispeakc0de @jimbugwadia @juntao @kannon92 @kkourt @kranurag7 @kyguy @leecalcote @michi-covalent @nate-double-u @pierDipi @ppatierno @q82419 @q82420 @qweeah @realshuting @rohit-nayak-ps @saswatamcode @skonto @slashben @suprjinx @vanshBhatia-A4k9 @yizha1 @yongtang @yurishkuro

**Reply by q82419** [reply]
Hi @aliok ,

Due to some technical issues, my email in LFX profile is still not updated to the mentor account info.
Can you help to invite again with email q82419 at gmail dot com ?
Thanks!

**Reply by q82419** [reply]
Hi @aliok  @nate-double-u ,

I found that the LFX profile updated and my email is corrected.
But I didn't be added as the mentor of [this](https://mentorship.lfx.linuxfoundation.org/project/74cecdf7-e886-4830-8bb0-7814f0d1aa2d) and [this](https://mentorship.lfx.linuxfoundation.org/project/4a8a4f26-0ca9-4517-8cce-582c92092e33) projects.
Can you help for this issue?
Thanks!

**Reply by nate-double-u** [reply]
Hi @q82419, It looks like your profile on those projects is the one attached to the email provided in the [project proposals](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug). So I'm not sure why you wouldn't have been invited to the system.

Could you open a support ticket here: https://jira.linuxfoundation.org/plugins/servlet/theme/portal/4/create/131
(I don't have visibility into the LFX backend, but the folks at support can help)

**Reply by q82419** [reply]
Hi @nate-double-u ,

Nevermind, I opened the support ticket and they helped to sync the email in my LFX profile 2 days ago.
But you sent the invitation last week, so I've been added as the mentor after you re-send the invitation.
Thanks!

**Reply by nate-double-u** [reply]
Thanks for letting me know. I'm glad things are sorted.

**Comment by kunal00000** [comment]
Hey @aliok, 

Is there a way we can specify our priority for projects.
Like if I apply to 3 projects then can I specify which project I want to get selected as my priority.

**Reply by asmitbm** [reply]
Hi @kunal00000 

No, you can’t specify the priority for the projects. However if you get selected for multiple projects, you can only select only one project of your choice and withdraw from the others. More information can be found in a [blog post](https://medium.com/@mehabhalodiya/experience-of-lfx-mentorship-for-cncf-kubernetes-673417f125c8) by an LFX Mentee in a similar position.

**Comment by ppatierno** [comment]
@nate-double-u @aliok I think I made a mistake when creating my LFX mentor profile not selecting the right project I should be the mentor for. I noticed this issue just today, is it possible to move forward and accept my request please? The project is this one (where it shows only Kyle as mentor and not me).

https://mentorship.lfx.linuxfoundation.org/project/8d301adf-94d8-4e5d-821d-f904ed15c3f9

**Reply by antonio-pedro99** [reply]
@ppatierno how can I dm you? I am interested in this project and I would like to have more info from you

**Reply by ppatierno** [reply]
@nate-double-u @aliok reading previous posts I think the problem is a different email. I am using my LFX profile with my personal email (a gmail one) but the invitation to be a mentor came to the email listed in the project proposal (a redhat one). Because I want to use my current LFX profile (with personal email), how can I sync them? I just opened a ticket to LFX.

**Reply by ppatierno** [reply]
@antonio-pedro99 you can use the email listed in the project proposals

**Reply by nate-double-u** [reply]
@ppatierno opening a ticket with the LFX is a good path. If you want to share the email you'd prefer to use I can also make the update. If you'd like you can dm me in the CNCF slack: @nate-double-u.

**Reply by ppatierno** [reply]
@nate-double-u I am in now, thanks!

**Comment by MariamFahmy98** [comment]
@nate-double-u @aliok I created my mentor profile and I requested to mentor Kyverno: validating admission policy phase 2. However my request hasn't been accepted yet.

**Reply by nate-double-u** [reply]
I've just added you -- you should get an email shortly.

**Comment by kannon92** [comment]
Can I get some help on https://mentorship.lfx.linuxfoundation.org/project/73d90321-62b3-498e-bf37-d899ec99df9e?  I noticed that I added Geoffrey Wilson and I as mentors for this project but Geoffrey is the only one listed.  

Is it possible to add me to the project list also?

**Reply by nate-double-u** [reply]
Hi @kannon92, could you please reach out to me on the CNCF slack: @nate-double-u. I have you listed twice on that project.

**Comment by ppatierno** [comment]
While reviewing the mentee list, I noticed that they are just providing a resume and cover letter.
Compared to what I am used to with GSoC, the process sounds to be a little bit weak.
With GSoC, the students have to provide a proposal where they describe the design and the solution they have in mind. It's a way for providing feedback and see how the student reacts in order to understand how much he/she is keen to work, how much understanding, etc etc. It's difficult picking a good candidate with just a resume and cover letter.
How others are dealing with this? I see you can add a "Task" (but it seems to be only for accepted meente?). It would be cool to add a task due to in a couple of days to get more tech-oriented proposals before making the right choice by May 29th.

@nate-double-u @aliok any thoughts on this?

**Reply by asmitbm** [reply]
> While reviewing the mentee list, I noticed that they are just providing a resume and cover letter. Compared to what I am used to with GSoC, the process sounds to be a little bit weak. With GSoC, the students have to provide a proposal where they describe the design and the solution they have in mind. It's a way for providing feedback and see how the student reacts in order to understand how much he/she is keen to work, how much understanding, etc etc. It's difficult picking a good candidate with just a resume and cover letter. How others are dealing with this? I see you can add a "Task" (but it seems to be only for accepted meente?). It would be cool to add a task due to in a couple of days to get more tech-oriented proposals before making the right choice by May 29th.
> 
> @nate-double-u @aliok any thoughts on this?

I believe there was an option to add additional tasks during the application period. For technical writing projects and design projects, an additional task was required to showcase samples of their work. The same approach could potentially be implemented for other projects as well.

Now, it can be quite challenging to assess mentees based solely on their resumes and cover letters. So, in my experience, for ORAS projects, genuinely interested mentees joined the community channels and directly approached the mentors on Slack to initiate discussions about project ideas and requested reviews of their proposals. This allowed us to provide feedback based on their understanding and approach. Additionally, we also assigned them a few tasks to complete before the application deadline.

This might seem unfair to students who are unsure how to approach mentors, but it also shows if they truly want to be part of the mentorship program and aren't just sending the same cover letters everywhere. But I do think there should be an option for proposals as well.

**Reply by haanhvu** [reply]
Here's how Thanos and Jaeger deal with this:

- Thanos: Shortlist the applicants, then ask further questions via email, and/or interview ([link](https://thanos.io/tip/contributing/mentorship.md/))
- Jaeger: Recommend applicants to include the proposal in the cover letter ([link](https://github.com/jaegertracing/jaeger/issues/4453))

**Reply by aliok** [reply]
(wearing my mentor hat)
Thanks for the feedback @haanhvu . This is actually a good idea. Also, I like how asking for a proposal is embedded in the upstream ticket. Genuinely interested applicants will notice that.

**Reply by chipzoller** [reply]
What I have done for my mentorship with the Kyverno project is similar to a proposal but I have framed it as a "challenge" instead which must be answered in the cover letter and in an external git repository. You can see the issue where this is spelled out [here](https://github.com/kyverno/policies/issues/546/). I have decided to take this route because I have historically not had good results in my mentees with these LFX mentorships. In the response, I'm searching for these qualities:

* Can the applicant follow directions? In many cases, they don't even acknowledge the challenge, they just submit résumé and cover letter. They are immediately eliminated. Mentees which cannot follow basic directions will cost more time and frustration than the output they'll deliver, and it'll typically lead to a poor experience for both.
* Can they self-start? The tool being used here is esoteric and so almost no one will have prior knowledge. I'm not looking for prior experience, I'm looking specifically for if they can get up and running by themselves and into a functional state without much (preferably any) hand-holding. This is important because, at this stage, I have already provided a tutorial on how to install and use the tool and pointed to a repository of our own with hundreds of examples. An ideal mentee will have enough to go on.
* Can they think critically? Writing great tests is about critical thinking in all the ways something can be bent and abused, and then covering each of those scenarios. Basic candidates will be able to think of the positive side of things ("Does it work when it should?"), but the ideal candidates also exercise the negative path(s) as well ("Does it work when it shouldn't?"). This is probably the most desirable criteria I want to see because it means they'll be able to grasp the concept and evolve in the desired direction, and typically in the time span required for this type of mentorship.
* Can they be technically accurate? This is the least important as it is directly proportional to time and experience with the tool, which, as stated will not be an expectation. But in this area, I still want to see their level of detail. This is detail from the perspective of not just bugs but optimum expression. Things that don't work are obviously problematic, but things which do work not that great are also not highly desired. I want to see here that they doing things with high quality.

**Reply by nate-double-u** [reply]
@ppatierno, there is a way to ask for extra files in the LFX system, and i believe that we can also link out to coding challenges (though I haven't done that yet, so we'd have to discuss how it would work). This should be included in the proposal PR (either in the discussion around the proposal, or in the proposal itself).

I'll make an update to the proposal instructions to help make it clear that proposals can include more than just a cover letter and resume.

**Reply by ppatierno** [reply]
I was new with LFX mentoring program but I had experience in GSoC as mentor, tbh I didn't know that LFX mentoring was "lacking" from this perspective, I mean that the process itself didn't take a deeply design proposal into account but instead just a resume and cover letter. Giving the freedom to the different communities to ask what they want is ok, but in any case I think that LFX mentoring should have an higher bar as a baseline. 
This time I went by pinging the few candidates I thought they could have been good ones and asked them for proposals.
Over this weekend there was a forth and back with them, with comments, feedback, discussions and it was good.

**Reply by yurishkuro** [reply]
Having gone through my first LFX program applications as a mentor just now, it feels like a proposal should be a table-stakes requirement at the CNCF level, with some standard minimum guidelines/questions. On my project only 15% of applications had proposals, even though we stated it in the upstream tickets - next time I will be mentioning them directly in the project description in this repo.

I also think the schedule is rather rushed. With GSoC we had applicants trying to work on `good-first-issue` tickets, but there's no time for that in LFX program.

**Comment by arkajyotiMukherjee** [comment]
I was added as a mentor with an email id that is not my primary id in my CNCF account. So my name is not showing up in this project https://mentorship.lfx.linuxfoundation.org/project/983193ea-9cca-405f-baa5-e6ade4df1ba2
I have raised a support ticket in JIRA as well. Would someone be able to help me here?

**Reply by TerryHowe** [reply]
Same

**Reply by nate-double-u** [reply]
Could you both reach out to me in the CNCF Slack (@nate-double-u), and provide me the email you use with your LFID? (or which you would prefer to use in this case)

**Comment by achiverram28** [comment]
@nate-double-u when can we expect the results of this terms lfx mentorship ? As in the repo , it is given that today is the last day for the review

**Reply by nate-double-u** [reply]
The deadline for selections is 5:00 pm PDT, then we need to do some paperwork. I'd expect results in the next day or so.

**Comment by eddycharly** [comment]
@nate-double-u I wanted to validate @VedRatan for https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug#cleanup-policies-phase-2. Am I too late ?

**Reply by nate-double-u** [reply]
Please use the form sent to the mentors to make your selection. If you haven't received that email, please DM me in the CNCF slack (@nate-double-u).

**Reply by eddycharly** [reply]
I submitted the form, thanks 🤞

**Comment by ppatierno** [comment]
So it seems that the mentee we chose is shown on the project page on the LFX website. He also contacted us directly to mention he has got an email from LFX for being accepted. We, as mentors, didn't get any. Can we assume that the project is now funded and we can start the mentoring?

@nate-double-u @aliok any insights ?

**Reply by nate-double-u** [reply]
Hi @ppatierno, I'll be sending a semester launch email tomorrow, and the [semester starts on June 1](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug#timeline), so you're welcome to reach out to your mentee and start welcoming them into the community!

**Comment by satyampsoni** [comment]
The project shows no mentee found which I have applied for. What does that mean?

**Reply by nate-double-u** [reply]
Which project? Most projects now should have selections made (there are a couple still going through their process). But most of the projects should have a mentee listed at this point.
